cli/sql.go

```diff
@@ -49,22 +56,52 @@ func runTerm(cmd *cobra.Command, args []string) {
 		_ = liner.Close()
 	}()
 
+	fmt.Fprint(osStdout, infoMessage)
+
+	// TODO(marc): allow passing statements on the command line,
+	// or specifying a flag. This would make testing much simpler.
+	// TODO(marc): add test.
+	var stmt string
 	for {
-		// TODO(marc): handle multi-line, this will require ';' terminated statements.
-		l, err := liner.Prompt("> ")
+		prompt := "> "
+		if len(stmt) != 0 {
+			// We're continuing a statement: indent relative to beginning.
+			prompt = "    "
+		}
+
+		l, err := liner.Prompt(prompt)
 		if err != nil {
 			if err != io.EOF {
-				fmt.Fprintf(os.Stderr, "Input error: %s\n", err)
+				fmt.Fprintf(osStderr, "Input error: %s\n", err)
 			}
 			break
 		}
-		if len(l) == 0 {
+
+		// See if we have a semicolon at the end of the line (ignoring
+		// whitespace after it).
+		tmpLine := strings.TrimSpace(l)
+		if len(tmpLine) == 0 {
+			continue
+		}
+
+		// Always use the non-trimmed line, we wouldn't want to
+		// break a statement due to a stripped begin or end space.
+		// We always insert a space when a newline is generated.
+		// This means that qualified names cannot be broken up.
+		stmt += " " + l
```

Comment:
Note that newlines are significant between string constants. That is, there is a difference between 

```
SELECT 'a' 'b'
```

and 

```
SELECT 'a'
'b'
```

The former is a syntax error while the latter concatenates the strings. So silly, but that is the standard.